src/components/Header/index.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from "react";
+import { usePrefs } from "../../prefs-context";
+import './style.css';
+
+const Header = () => {
+
+    const {changePrefs} = usePrefs();
+    const [veganPref, setVeganPref] = useState("All available")
+    
+
+    const handleSelect = (e) => {
+        
+        setVeganPref(e.target.value);
+        if (e.target.value === "Vegan only") {
+            changePrefs(true)}
+            else changePrefs(false);
+        }
+
+    return (
+        <header>
+                <span>Choose your pizza topping preferences: </span>
+                <select className="select" value={veganPref} onChange={handleSelect}>
+                <option>All available</option>
+               <option>Vegan only</option>
+                </select>
```

Comment:
```

    const {changePrefs, veganOnly} = usePrefs();
    const veganPref = veganOnly ? "vegan" : "all";
...
                <select className="select" value={veganPref} onChange={handleSelect}>
                <option value="all">All available</option>
               <option value="vegan">Vegan only</option>
                </select>
```
takhle by to mělo fungovat už :)

Reply:
Tady chceš aby se ti defaultní hodnota řídila podle stavu "veganOnly". Jde o to že takhle máš jednu logickou informaci máš vyjádřenou na dvou místech v kódu, a když budeš chtít změnit co je defaultně zobrazeného, musíš změnit 2 místa. To je typicky vlastnost kterou v aplikaci nechceš protože je vysoký riziko bugů - buď ty nebo kolega zapomenete že se něco musí změnit na 2 místech....

Reply:
Jasný, na tom jsem se hodně zasekla, vůbec jsem nevěděla, jak ten select vyřešit, aby se řídil podle stavu veganOnly, tak jsem to nakonec udělala takhle nešikovně...